src/local/motion/NonRigidObjectTransformationUsingBSplines.cxx

```diff
@@ -340,6 +406,46 @@ NonRigidObjectTransformationUsingBSplines()
   this->set_defaults();
 }
 
+template <int num_dimensions, class elemT>
+NonRigidObjectTransformationUsingBSplines<num_dimensions,elemT>::
+NonRigidObjectTransformationUsingBSplines(const std::string &filename_x, const std::string &filename_y, const std::string &filename_z, const int bspline_order)
+{
+    this->set_defaults();
+
+    // Get the origin and spacing from the x and assume it's the same for y and z
+    shared_ptr<DiscretisedDensity<3,float> > image_sptr(read_from_file<DiscretisedDensity<3,float> >(filename_x));
+
+    if (is_null_ptr(image_sptr))
+        error("Error reading %s", filename_x.c_str());
+
+    VoxelsOnCartesianGrid<float> const * voxels_ptr = dynamic_cast<VoxelsOnCartesianGrid<float> const *>(image_sptr.get());
+
+    if (is_null_ptr(voxels_ptr))
+        error(boost::format("Error reading %1%: should be of type VoxelsOnCartesianGrid") % filename_x);
+
+    this->_origin       = image_sptr->get_origin();
+    this->_grid_spacing = voxels_ptr->get_grid_spacing();
+
+    this->_deformation_field_from_file_x = filename_x;
+    this->_deformation_field_from_file_y = filename_y;
+    this->_deformation_field_from_file_z = filename_z;
```

Comment:
the filenames are not going to be defined all the time, but in any case, let's just set `this->_deformation_field_multicomponent_filename=""` here